src/credentials/examples/DeviceAttestationVerifierExample.cpp

```diff
@@ -0,0 +1,194 @@
+
+    return CopySpanToMutableSpan(ByteSpan{ sPAALookupTable[paaLookupTableIdx].mPAACertificate }, out_dac_buffer);
+}
+
+class ExampleDACVerifier : public DeviceAttestationVerifier
+{
+public:
+    AttestationVerificationResult
+    VerifyAttestationInformation(const ByteSpan & attestation_info_buffer, const ByteSpan & attestation_challenge_buffer,
+                                 const ByteSpan & attestation_signature_buffer, const ByteSpan & pai_cert_der_buffer,
+                                 const ByteSpan & dac_cert_der_buffer, const ByteSpan & attestation_nonce) override;
+};
+
+AttestationVerificationResult ExampleDACVerifier::VerifyAttestationInformation(const ByteSpan & attestation_info_buffer,
+                                                                               const ByteSpan & attestation_challenge_buffer,
+                                                                               const ByteSpan & attestation_signature_buffer,
+                                                                               const ByteSpan & pai_cert_der_buffer,
+                                                                               const ByteSpan & dac_cert_der_buffer,
+                                                                               const ByteSpan & attestation_nonce)
+{
+    // match DAC and PAI VIDs
+    if (!pai_cert_der_buffer.empty())
+    {
+        VendorId paiVid;
+        VendorId dacVid;
+
+        VerifyOrReturnError(ExtractVIDFromX509Cert(pai_cert_der_buffer, paiVid) == CHIP_NO_ERROR,
+                            AttestationVerificationResult::kPaiFormatInvalid);
+        VerifyOrReturnError(ExtractVIDFromX509Cert(dac_cert_der_buffer, dacVid) == CHIP_NO_ERROR,
+                            AttestationVerificationResult::kDacFormatInvalid);
+
+        VerifyOrReturnError(paiVid == dacVid, AttestationVerificationResult::kDacVendorIdMismatch);
+    }
+
+    uint8_t akidBuf[Credentials::kKeyIdentifierLength];
+    MutableByteSpan akid(akidBuf);
+    ExtractAKIDFromX509Cert(pai_cert_der_buffer.empty() ? dac_cert_der_buffer : pai_cert_der_buffer, akid);
+
+    constexpr size_t paaCertAllocatedLen = kMaxDERCertLength;
+    chip::Platform::ScopedMemoryBuffer<uint8_t> paaCert;
+
+    VerifyOrReturnError(paaCert.Alloc(paaCertAllocatedLen), AttestationVerificationResult::kNoMemory);
+
+    MutableByteSpan paa(paaCert.Get(), paaCertAllocatedLen);
+    VerifyOrReturnError(GetProductAttestationAuthorityCert(akid, paa) == CHIP_NO_ERROR,
+                        AttestationVerificationResult::kPaaNotFound);
+
+    VerifyOrReturnError(ValidateCertificateChain(paa.data(), paa.size(), pai_cert_der_buffer.data(), pai_cert_der_buffer.size(),
+                                                 dac_cert_der_buffer.data(), dac_cert_der_buffer.size()) == CHIP_NO_ERROR,
+                        AttestationVerificationResult::kPaaUntrusted);
```

Comment:
The chain failing has to account for which item in the chain failed. If it's unknown which certificate failed validation, then we cannot be clear with the errors. For now, I would say that we should use `kDacSignatureInvalid` for this failure. Other implementations of this code can use something with better error reporting than `ValidateCertificateChain` which can identify what is the element that failed.

I also think kPaaUntrusted/kPaaNotFound are equivalent, so we should pick one value.

Reply:
will update to use kDacSignatureInvalid